syscall.go

```diff
@@ -178,21 +193,115 @@ func waitForBlockDevice(deviceName string) error {
 	return nil
 }
 
-func mountContainerRootFs(containerID, image, rootFs, fsType string) (string, error) {
+// scanSCSIBus scans SCSI bus for the given SCSI address(SCSI-Id and LUN)
+func scanSCSIBus(scsiAddr string) error {
+	scsiHostPath := "/sys/class/scsi_host"
+	if _, err := os.Stat(scsiHostPath); err != nil {
+		return err
+	}
+
+	files, err := ioutil.ReadDir(scsiHostPath)
+	if err != nil {
+		return err
+	}
+
+	tokens := strings.Split(scsiAddr, ":")
+	if len(tokens) != 2 {
+		return fmt.Errorf("Unexpected format for SCSI Address : %s, expect SCSIID:LUN", scsiAddr)
+	}
+
+	// Scan scsi host passing in the channel, SCSI id and LUN, channel is always 0
+	scanData := []byte(fmt.Sprintf("0 %s %s", tokens[0], tokens[1]))
+
+	for _, file := range files {
+		host := file.Name()
+		scanPath := filepath.Join(scsiHostPath, host, "scan")
+		if err := ioutil.WriteFile(scanPath, scanData, 0666); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// findSCSIDisk finds the SCSI disk name associated with the given SCSI address
+// This approach eliminates the need to predict the disk name on the host side,
+// but we do need to rescan SCSI bus for this.
+func findSCSIDisk(scsiAddr string) (string, error) {
+	// Here in "0:0", the first number is the SCSI host number, while the second number is always 0.
+	scsiPath := fmt.Sprintf("/sys/class/scsi_disk/0:0:%s/device/block", scsiAddr)
+
+	if _, err := os.Stat(scsiPath); err != nil {
+		return "", err
+	}
+
+	files, err := ioutil.ReadDir(scsiPath)
+	if err != nil {
+		return "", err
+	}
+
+	if len(files) > 1 {
```

Comment:
There is an implicit assumption that a single file was read, so how about the following:

```
l := len(files)

if l == 0 || l > 1 {
    ...
}
```